app/src/main/java/com/example/travellens/MessageActivity.java

```diff
@@ -63,25 +74,58 @@ protected void onCreate(Bundle savedInstanceState) {
         // get intent with the user info of who you're sending to
         intent = getIntent();
         post = intent.getExtras().getParcelable("post");
+
         ibSend.setOnClickListener(this::sendMessage);
+        bDeleteAttachment.setOnClickListener(this::deleteAttachmentFromPost);
+
         adaptMessages(ParseUser.getCurrentUser().getString(Post.KEY_FIREBASE_USER_ID), post.getUser().getString(Post.KEY_FIREBASE_USER_ID));
+        attachWidgets();
+
 
+    }
+
+    private void attachWidgets() {
         Glide.with(getApplicationContext())
-                .load(post.getUser().getParseFile(Post.KEY_PROFILE_PICTURE))
+                .load(post.getUser().getParseFile(Post.KEY_PROFILE_PICTURE).getUrl())
                 .circleCrop()
                 .into(ivReceiverPicture);
         tvReceiverUsername.setText(post.getUser().getUsername());
         tvReceiverName.setText(post.getUser().getString(Post.KEY_NAME));
+        if (post == null) {
+            ivPictureFromPost.setVisibility(View.GONE);
+        } else {
+            Glide.with(getApplicationContext()).load(post.getParseFile(Post.KEY_IMAGE).getUrl()).centerCrop().transform(new RoundedCorners(16)).into(ivPictureFromPost);
+            attachPhotoToMessage = true;
+        }
+    }
+
+    private void deleteAttachmentFromPost(View view) {
+        ivPictureFromPost.setVisibility(View.GONE);
+        attachPhotoToMessage = false;
     }
 
     private void sendMessage(View view) {
         reference = FirebaseDatabase.getInstance().getReference();
 
+        Date date = new Date();
+        SimpleDateFormat formatter = new SimpleDateFormat("MMM dd hh:mm a");
+        String dateAndTime = formatter.format(date);
+
         HashMap<String, Object> hashMap = new HashMap<>();
         hashMap.put("sender", ParseUser.getCurrentUser().getString(Post.KEY_FIREBASE_USER_ID));
         hashMap.put("receiver", post.getUser().getString(Post.KEY_FIREBASE_USER_ID));
         hashMap.put("message", etMessage.getText().toString());
+        hashMap.put("dateAndTime", dateAndTime);
+        if (post != null && attachPhotoToMessage == true) {
+            // attach image to message
+            hashMap.put("photo", post.getParseFile(Post.KEY_IMAGE).getUrl());
+        } else {
+            hashMap.put("photo", null);
```

Comment:
Do you need this put with a null value while adding the message?

Reply:
Actually even for code readability, it's not really needed. The if condition itself already lets us know that the photo is only added if there is a photo available to add.